app/controllers/flights_controller.rb

```diff
@@ -4,10 +4,25 @@ def show
   end
 
   def index
-    @trips = SearchFacade.get_flights(flight_params)
-    if @trips.class == String
-      flash[:error] = @trips
-      redirect_to root_path
+    if params[:commit] == 'Search Locations'
+      @trips = SearchFacade.get_flights(flight_params)
+      if @trips.class == String
+        flash[:error] = @trips
+        redirect_to root_path
+        return
+      end
+      @trips.each do |trip|
+        Rails.cache.write(trip.flight_id, trip)
```

Comment:
Nice add of a cache

Reply:
Nice use of caching! If we'd like we can also pass in an additional argument like `expires_in` to set a TTL for the cached item.

Reply:
Good job implementing caching